src/Lykke.Job.CandlesHistoryWriter.Services/HistoryMigration/CandleExtensions.cs

```diff
@@ -38,8 +38,8 @@ public static ICandle ExtendBy(this ICandle self, ICandle newCandle)
                 selfIsOlder ? newCandle.Close : self.Close,
                 Math.Max(self.High, newCandle.High),
                 Math.Min(self.Low, newCandle.Low),
-                self.TradingVolume + newCandle.TradingVolume,
-                self.TradingOppositeVolume + newCandle.TradingOppositeVolume,
+                Convert.ToDouble((decimal)self.TradingVolume + (decimal)newCandle.TradingVolume), // It's a bit messy, but really allows to keep precision in addiction operation.
```

Comment:
Not sure if this will help. Precision issue is with representation of the double itself, not in summation of the double